src/validation.cpp

```diff
@@ -1295,36 +1295,41 @@ bool MemPoolAccept::ConsensusScriptChecks(const ATMPArgs& args, Workspace& ws)
     return true;
 }
 
-bool MemPoolAccept::FinalizeSubpackage(const ATMPArgs& args, std::vector<Workspace>& workspaces)
+void MemPoolAccept::FinalizeSubpackage(const ATMPArgs& args)
 {
     AssertLockHeld(cs_main);
     AssertLockHeld(m_pool.cs);
-    const CTransaction& tx = *workspaces.front().m_ptx;
-    const uint256& hash = workspaces.front().m_hash;
-    TxValidationState& state = workspaces.front().m_state;
-    const bool bypass_limits = args.m_bypass_limits;
 
     if (!m_subpackage.m_all_conflicts.empty()) Assume(args.m_allow_replacement);
     // Remove conflicting transactions from the mempool
     for (CTxMemPool::txiter it : m_subpackage.m_all_conflicts)
     {
-        LogDebug(BCLog::MEMPOOL, "replacing mempool tx %s (wtxid=%s, fees=%s, vsize=%s). New tx %s (wtxid=%s, fees=%s, vsize=%s)\n",
+        LogDebug(BCLog::MEMPOOL, "replacing mempool tx %s (wtxid=%s, fees=%s, vsize=%s). ",
                 it->GetTx().GetHash().ToString(),
                 it->GetTx().GetWitnessHash().ToString(),
                 it->GetFee(),
-                it->GetTxSize(),
-                hash.ToString(),
-                tx.GetWitnessHash().ToString(),
-                workspaces[0].m_tx_handle->GetFee(),
-                workspaces[0].m_tx_handle->GetTxSize());
+                it->GetTxSize());
+        FeeFrac feerate = m_subpackage.m_changeset->GetAggregateFeeRate();
```

Comment:
You could use `m_subpackage.m_total_modified_fees, m_subpackage.m_total_vsize` here? Unless you feel that `CTxMemPoolChangeSet::GetAggregateFeeRate` will be useful in the future.

Reply:
Thanks that is much better; the only reason I had introduced `GetAggregateFeeRate()` was for access in this one place, so now that is gone.